btree/btree.h

```diff
@@ -120,6 +141,25 @@ void m0_btree_op_credit(const struct m0_btree_op *bt,
 int  m0_btree_mod_init(void);
 void m0_btree_mod_fini(void);
 
+
+/**
+ * This macro calls the 'action' function and follows it by calling
+ * m0_sm_op_tick() to execute the state machine.
+ *
+ * IMPORTANT: The 'action' routine should execute the call m0_sm_op_init() for
+ * setting the *_tick() function which is eventually be called by
+ * m0_sm_op_tick() function.
+ */
+#define M0_BTREE_OP_SYNC_WITH(op, action)       \
+	({                                      \
+		struct m0_sm_op *__opp = (op);  \
+						\
```

Comment:
We might need to consider following functions before action in this macro:
m0_sm_group_lock()
m0_sm_op_exec_init()

These must be called before action so that the tick function related to action will work properly.

Reply:
Please make the exact changes that you need since you are already implementing the create_tree function.

Reply:
Ok.
I will update it the PR tree-create PR #668